src/hb-ot-layout-gsubgpos.hh

```diff
@@ -83,7 +92,9 @@ struct hb_closure_context_t :
     if (is_lookup_done (lookup_index))
       return false;
 
-    done_lookups->set (lookup_index, glyphs->get_population ());
+    /* a lookup might be visited twice from different rules/starting glyphs, while c->glyphs is
+     * only updated with output glyphs at the end of recursion*/
+    done_lookups->set (lookup_index, glyphs->get_population () + output->get_population ());
```

Comment:
I think there are some cases where this doesn't work. Consider a case where you recurse into a look with a specific set of parent glyphs. No additional glyphs are added to output. Then in the same iteration with a different set of parent glyphs you go back to the same lookup. It would be treated as already being done since output hasn't changed.

Unfortunately I think we're going to have to switch to a more complicated done_lookups value, that actually records the specific set of glyphs  that have been used so far. Similar to what fontTools does: https://github.com/fonttools/fonttools/blob/master/Lib/fontTools/subset/__init__.py#L1176